### PR TITLE
Make drawing bbox an option in showAnns.

### PR DESCRIPTION
**Pull request by wenting-zhao**

### CODE REVIEW

**Comment by pliniosilveira** [comment]
Thanks for that! :)

**Review by pliniosilveira — APPROVED** [review]

**Comment by guysoft** [comment]
Hey, any reason this is not merged?

File: PythonAPI/pycocotools/coco.py [hunk 1]
```diff
@@ -286,6 +286,14 @@ def showAnns(self, anns):
                             plt.plot(x[sk],y[sk], linewidth=3, color=c)
                     plt.plot(x[v>0], y[v>0],'o',markersize=8, markerfacecolor=c, markeredgecolor='k',markeredgewidth=2)
                     plt.plot(x[v>1], y[v>1],'o',markersize=8, markerfacecolor=c, markeredgecolor=c, markeredgewidth=2)
+
+                if draw_bbox:
+                    [bbox_x, bbox_y, bbox_w, bbox_h] = ann['bbox']
+                    poly = [[bbox_x, bbox_y], [bbox_x, bbox_y+bbox_h], [bbox_x+bbox_w, bbox_y+bbox_h], [bbox_x+bbox_w, bbox_y]]
+                    np_poly = np.array(poly).reshape((4,2))
+                    polygons.append(Polygon(np_poly))
+                    color.append(c)
```
Comment (wooridle) [inline PythonAPI/pycocotools/coco.py]: I think the color needs to change for visibility.

Comment (CSU-NXY) [reply PythonAPI/pycocotools/coco.py]: How about using dashing line?
```suggestion
ax.add_patch(Polygon(np_poly, linestyle='--', facecolor='none', edgecolor=c, linewidth=2))
```
With which we can have folling output
sample

**Comment by codeislife99** [comment]
Hi, can this branch be merged please ?

**Comment by panovr** [comment]
Hope this branch willl be merged.

**Comment by ividal** [comment]
Upvoting for this to be merged. ShowAnns with bboxes definitely needed.

**Comment by tloki** [comment]
Will this ever get merged? Or at least reject it so we don't count on in in the future...

**Comment by nsarang** [comment]
More than a year since this PR was made. This is just lazy.

**Comment by Yudezhi** [comment]
mark for subscribe

**Comment by gautamchitnis** [comment]
please merge this branch :(

**Comment by nicpayne713** [comment]
Please merge this - it's a great tool. We only need bounding boxes and not segmentation - this simplifies a lot of reporting we do on our modeling!

**Review by tylin — APPROVED** [review]
This looks good to me! Thanks everyone!

**Comment by pliniosilveira** [comment]
I can't express how happy I'm to finally have it merged. :)

**Comment by gautamchitnis** [comment]
For anyone looking to use this commit in Windows, check out my fork of philferriere's repo: [https://github.com/gautamchitnis/cocoapi/tree/cocodataset-master](https://github.com/gautamchitnis/cocoapi/tree/cocodataset-master)

File: PythonAPI/pycocotools/coco.py (hunk 1 above)
Comment (panovr) [reply PythonAPI/pycocotools/coco.py:295]: > How about using dashing line?
> 
> ```
> ax.add_patch(Polygon(np_poly, linestyle='--', facecolor='none', edgecolor=c, linewidth=2))
> ```
> 
> With which we can have following output
> sample

@CSU-NXY where should this line be put?

**Pull request merged**